plugins/scaffolder/src/components/TemplatePage/TemplatePage.tsx

```diff
@@ -120,7 +121,12 @@ export const TemplatePage = ({
   const navigate = useNavigate();
   const rootLink = useRouteRef(rootRouteRef);
   const { schema, loading, error } = useTemplateParameterSchema(templateName);
-  const [formState, setFormState] = useState({});
+  const query = qs.parse(window.location.search, {
+    ignoreQueryPrefix: true,
+  });
+  const [formState, setFormState] = useState(
+    (query.formData ?? {}) as Record<string, any>,
```

Comment:
this should be a function arg to useState, that does the qs.parse - so it just runs once at startup

Reply:
good point! updated.